scalability_and_performance/enabling-workload-partitioning.adoc

```diff
@@ -0,0 +1,88 @@
+
+toc::[]
+
+In resource-constrained environments, you can use workload partitioning to isolate {product-title} services, cluster management workloads, and infrastructure pods to run on a reserved set of CPUs.
+
+The minimum number of reserved CPUs required for the cluster management is four CPU Hyper-Threads (HTs).
+With workload partitioning, you annotate the set of cluster management pods and a set of typical add-on Operators for inclusion in the cluster management workload partition.
+These pods operate normally within the minimum size CPU configuration.
+Additional Operators or workloads outside of the set of minimum cluster management pods require additional CPUs to be added to the workload partition.
+
+Workload partitioning isolates user workloads from platform workloads using standard Kubernetes scheduling capabilities.
+
+The following is an overview of the configurations required for workload partitioning:
+
+* The `install-config.yaml` file requires an additional field: `cpuPartitioningMode`.
++
+[source,yaml]
+----
+apiVersion: v1
+baseDomain: devcluster.openshift.com
+cpuPartitioningMode: AllNodes <1>
+Compute:
+  - architecture: amd64
+    hyperthreading: Enabled
+    name: worker
+    platform: {}
+    replicas: 3
+controlPlane:
+  architecture: amd64  
+  hyperthreading: Enabled
+  name: master
+  platform: {}
+  replicas: 3
+----
+<1> Sets up a cluster for CPU partitioning at install time. The default value is `None`. 
++
+[NOTE]
+====
+Workload partitioning can only be enabled during cluster installation. You cannot disable workload partitioning post-installation.
+====
+
+* In the performance profile, specify the `isolated` and `reserved` CPUs. 
++
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile
+metadata:
+  name: openshift-node-workload-partitioning-worker
+spec:
+  cpu:
+    isolated: 0,1 <1>
+    reserved: "2-3" <2>
+  machineConfigPoolSelector:
+    pools.operator.machineconfiguration.openshift.io/worker: ""
+  nodeSelector:
+    node-role.kubernetes.io/worker: ""
+----
+<1> Sets the isolated CPUs. Ensure all of the Hyper-Threading pairs match.
+<2> Specifies the workloads to pin to the reserved `CPUSet`. When workload partitioning is enabled, platform containers and platform services such as `systemd`, `CRI-O`, and `Kubelet`  are restricted to these CPUs. All CPUs that are not isolated should be reserved.
+
+* The performance profile: 
++
+--
+** pins cluster services such as systemd and kubelet to the CPUs that are defined in the `spec.cpu.reserved` field
+** pins the {product-title} infrastructure pods to a defined `cpuset` configuration
+--
++
+[NOTE]
+====
+Using the Node Tuning Operator, you can configure the performance profile to also pin system-level apps for a complete workload partitioning configuration on the node.
+====
+
+* The CPUs that you specify in the performance profile `spec.cpu.reserved` field and the workload partitioning `cpuset` field must match.
```

Comment:
is the cpuset field part of another spec? or is it in `spec.cpu`? It may be helpful for users to place `cpuset`within the broader spec, as you've done with `spec.cpu.reserved`.

Reply:
Thanks. I've tidied this up based on discussion with SME prompted by this comment.